Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -49,38 +51,7 @@ namespace ROS2
         AZ_Assert(m_root, "SDF Root is nullptr");
         if (m_root != nullptr)
         {
-            VisualsMaker::MaterialNameMap materialMap;
-            auto GetVisualsFromModel = [&materialMap](const sdf::Model& model)
-            {
-                for (uint64_t linkIndex{}; linkIndex < model.LinkCount(); ++linkIndex)
-                {
-                    if (const sdf::Link* link = model.LinkByIndex(linkIndex); link != nullptr)
-                    {
-                        for (uint64_t visualIndex{}; visualIndex < link->VisualCount(); ++visualIndex)
-                        {
-                            if (const sdf::Visual* visual = link->VisualByIndex(visualIndex); visual != nullptr)
-                            {
-                                if (const sdf::Material* material = visual->Material(); material != nullptr)
-                                {
-                                    const std::string visualName = visual->Name();
-                                    materialMap.emplace(AZStd::string{ visualName.c_str(), visualName.size() }, material);
-                                }
-                            }
-                        }
-                    }
-                }
-            };
-
-            // Iterate over all visuals to get their materials
-            auto VisitAllModels = [&GetVisualsFromModel](const sdf::Model& model, const Utils::ModelStack&) -> Utils::VisitModelResponse
-            {
-                GetVisualsFromModel(model);
-                // Continue to visit all models within the SDF document and query their <visual> tags
-                return Utils::VisitModelResponse::VisitNestedAndSiblings;
-            };
-            Utils::VisitModels(*m_root, VisitAllModels);
-
-            m_visualsMaker = VisualsMaker(AZStd::move(materialMap), urdfAssetsMapping);
+            m_visualsMaker = VisualsMaker(urdfAssetsMapping);
```

Comment:
Actaully this entire if block can be removed.
We can pass in the urdfAssetMapping variable as part of the constructor initialization list for m_visualsMaker on line 43.

We had to make a new instance before to pass in the MaterialMap, but as we are no longer doing that, we can initialize it directly in the constructor.

Reply:
Fixed!